### ISSUE TITLE
Strings on options get cleared when the function properties window gets updated

### ISSUE THREAD

**Issue by wwwysocki** [Orckestra/CMS-Foundation#188]
**Prepare**:

1. Create a Razor function.
2. Use the code below.
3. Save it

```
@inherits RazorFunction

@functions {
    public override string FunctionDescription
    {
        get { return "A demo function that outputs a hello message."; }
    }

    [FunctionParameter(Label = "Content")]
    public XhtmlDocument Content { get; set; }


    [FunctionParameter(Label = "Options",
        DefaultValue = "String 1,String 3",
        WidgetMarkup = @"<f:widgetfunction xmlns:f=""http://www.composite.net/ns/function/1.0""
name=""Composite.Widgets.String.Selector"" label="""" bindingsourcename="""">
<f:helpdefinition xmlns:f=""http://www.composite.net/ns/function/1.0"" helptext="""" />
<f:param xmlns:f=""http://www.composite.net/ns/function/1.0"" name=""Options"">
<f:function xmlns:f=""http://www.composite.net/ns/function/1.0"" name=""Composite.Utils.String.Split"">
<f:param name=""String"" value=""String 1,String 2,String 3"" />
</f:function>
</f:param><f:param name=""Multiple"" value=""True"" /><f:param name=""Required"" value=""False"" /></f:widgetfunction>")]
    public string Options { get; set; }
}
```

**Repro**:

1. Insert the function on a page
2. Click in the "Content" parameter to open the Visual XHTML Editor.
3. Click OK in the Editor.
4. Check the "Options" parameter

**Expected**: The option check boxes have texts ("String 1" etc)
**Actual**: The texts on options get cleared (see below)

![option-texts-cleared](https://cloud.githubusercontent.com/assets/14821915/16149603/a96de9f2-349a-11e6-825e-c0923a00cf74.png)

**Comment by neexite** [comment]
server returns empty "label" attributes for checkboxes on postback

**Comment by napernik** [comment]
Fixed in "fix-bugs" branch

**Comment by wwwysocki** [comment]
Verified in "fix-bugs" (20161003.4)
